spec/migrations/20170529142557_unify_container_definition_and_container_spec.rb

```diff
@@ -0,0 +1,124 @@
+require_migration
+
+describe UnifyContainerDefinitionAndContainer do
+  let(:container_stub) { migration_stub(:Container) }
+  let(:container_definition_stub) { migration_stub(:ContainerDefinition) }
+
+  let(:security_context_stub) { migration_stub(:SecurityContext) }
+  let(:container_env_var_stub) { migration_stub(:ContainerEnvVar) }
+  let(:container_port_config_stub) { migration_stub(:ContainerPortConfig) }
+
+  let(:definition_hash) do
+    {
+      :image              => "my_image",
+      :image_pull_policy  => "image_pull_policy",
+      :memory             => "memory",
+      :cpu_cores          => 1.1,
+      :container_group_id => 1,
+      :privileged         => true,
+      :run_as_user        => 2,
+      :run_as_non_root    => true,
+      :capabilities_add   => "capabilities_add",
+      :capabilities_drop  => "capabilities_drop",
+      :command            => "command"
+    }
+  end
+
+  let(:container_hash) do
+    {
+      :name       => "mycontainer",
+      :ems_ref    => "123",
+      :ems_id     => 1,
+      :old_ems_id => 1,
```

Comment:
Please take realistic data from real DB.  (that's not how ems_ref looks, not sure ems_id == old_ems_id == 1 is possible etc...)
And since I have some worries about the overlapping columns like `ems_ref`, consider spelling out full data for both AND the expected merge, without assuming `definition_hash.merge(container_hash))`.

[have I mentioned migrations scare me, perhaps unreasonably?]